include/unifex/awaitable_sender.hpp

```diff
@@ -155,12 +158,15 @@ struct _sender<Awaitable>::type {
                     };
                   }(co_await (Awaitable &&) awaitable);
             }
+
+#if !UNIFEX_NO_EXCEPTIONS
           } catch (...) {
             ex = std::current_exception();
           }
           co_yield[&] {
             unifex::set_error(std::move(receiver), std::move(ex));
           };
```

Comment:
I was worried the compiler wouldn't be able to elide this, or `ex`, if I just relied on `UNIFEX_TRY` and `UNIFEX_CATCH` here so I stuck with the previous approach.

Reply:
@lewissbaker Am I correct in assuming that the `co_yield`s in this coroutine are actually terminal? That is, they delete the coroutine?

Reply:
yes, the `co_yield` here is terminal.
It is used as a mechanism to invoke `set_error()` after the coroutine as suspended - so that it's valid for `set_error()` to then destroy the coroutine frame when the operation-state is destroyed.